src/map.h

```diff
@@ -18,52 +18,60 @@
 
 #include <stdbool.h>
 #include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+typedef struct map_node map_node_t;
+struct map_node {
+    void *key;
+    void *val;
+    struct {
+        map_node_t *left, *right_red;
+    } link;
+};
+
+typedef struct rb_tree {
+    map_node_t *root;
+    int (*cmp)(const map_node_t *, const map_node_t *);
+    size_t key_size;
+    size_t val_size;
+} map_internal_t;
```

Comment:
I didn't get the intention why you introduced `map_internal_t`. Can you write more consistent and simpler code?